openff/toolkit/utils/builtin_wrapper.py

```diff
@@ -123,14 +122,12 @@ def assign_partial_charges(
                 strict_n_conformers=strict_n_conformers,
             )
 
-        partial_charges = unit.Quantity(
-            np.zeros((molecule.n_particles)), unit.elementary_charge
-        )
         if partial_charge_method == "zeroes":
-            pass
+            partial_charges = unit.Quantity(
+                molecule.n_atoms * [0], unit.elementary_charge
+            )
         elif partial_charge_method == "formal_charge":
-            for part_idx, particle in enumerate(molecule.particles):
-                partial_charges[part_idx] = particle.formal_charge
+            partial_charges = [atom.formal_charge for atom in molecule.atoms]
```

Comment:
I'm a little cautious here. Because the previous code instantiated this with `np.zeros`, these would have set to float64s, so I'd like to make sure that we don't change these to ints in the new code if we don't have to. 

```suggestion
            partial_charges = [float(atom.formal_charge) for atom in molecule.atoms]
```